### ISSUE TITLE
DNS (Windows) defaults to 127.0.0.1 when connected to .ovpn

### ISSUE THREAD

**Issue by wellloaded** [Windscribe/Desktop-App#102]
I have specified in the settings: custom DNS (my own router IP as I need this) however as soon as the WSclient connects my external .ovpn connection the DNS server changes to 127.0.0.1 and doesn't resolve anything any more (since there's no local DNS service running on the host).

Is this a bug or a configuration issue? 

How to fix please? I really need the DNS to be set to 10.10.0.1 while connected to the VPN.

Thanks

(Using Guinea pig release, standard release doesn't have this).

**Comment by yegors** [comment]
Are you using Custom DNS server in Windscribe? 

![image](https://github.com/Windscribe/Desktop-App/assets/817935/4fbfd34c-6ae2-4b95-a048-624e69132b94)

**Comment by wellloaded** [comment]
Precisely and that's what it appears not to be working.
As mentioned DNS resolution (nslookup) changes server to 127.0.0..1 post installation.

**Comment by yegors** [comment]
When custom DNS it used, a DNS proxy service is spawned on that IP, which sends DNS queries to whatever Upstream 1 is set to. What are you setting it to?

**Comment by wellloaded** [comment]
My Custom DNS is actually pointing to my own (FreshTomato) LAN router. The reasons being:
- I have Stubby/dnscrypt and adblock running in there which I trust
- I need DNS resolution for my local intra-site VPN (tinc)

**Comment by wellloaded** [comment]
I'm sorry but I had to revert to stable to have this operational again so I can't test right now...

**Comment by yegors** [comment]
Well, if there is a bug here, we can't fix it without additional details / exact steps to reproduce.

**Comment by wellloaded** [comment]
I appreciate. What I did was:
- install guinea pig
- set custom dns to my router LAN IP
- verified default resolution is resolved by 127.0.0.1 (despite hardcoded setting on VPN nic)
- no actual dns resolution is performed

If I get some time I'll try to reinstall guinea pig and see if forcing nslookup to use my router LAN IP has any effect.

**Comment by wellloaded** [comment]
Setting nslookup manually to my router IP doesn't resolve either.

![image](https://github.com/Windscribe/Desktop-App/assets/46318040/cd0300f2-e63d-4558-ba62-4bf0500982bc)

The WS firewall is off. 
Disconnecting the VPN and repeating the nslookup works at first attempt.

**Comment by bernerdad** [comment]
Just a quick note that we will be working to resolve this issue for our next guinea pig release.  Thank you for bringing it to our attention!

**Comment by bernerdad** [comment]
Hi.  Here's an early preview of the v2.7.11 guinea pig release.  Please let us know if it resolves this issue for you.  Thanks!
https://deploy.totallyacdn.com/desktop-apps/2.7.11/Windscribe_2.7.11_guinea_pig.exe

**Comment by wellloaded** [comment]
I will try this ASAP. Before I do so, is there a safe way to completely remove the existing client (files, registry, etc)?
I think it would make sense to get a fresh start to test this.

Thanks

**Comment by wellloaded** [comment]
In the meantime I have installed the guinea_pig over the stable. Domains now do get resolved when I use my router LAN IP address as the Custom DNS server within the WS client. Two points though:

- Only one thing though... nslookup still uses 127.0.0.1 as the resolver, this hides from the user the actual resolver used by Windscribe so I can only assume it uses my router.
- how can I specify more than 1 Custom DNS server (e.g secondary DNS server available on the LAN) ? Would a comma (or space?) separated list work? Either ways this should be added as a comment on the GUI in my opinion.
Thanks

**Comment by bernerdad** [comment]
> I will try this ASAP. Before I do so, is there a safe way to completely remove the existing client (files, registry, etc)? I think it would make sense to get a fresh start to test this.
> 
> Thanks

To get a fresh start, run the installer and on its main screen click the gear icon at the bottom.  Toggle on the factory reset option and click the OK button.  That will delete everything created by a prior install of the app once the install process starts.

**Comment by bernerdad** [comment]
> In the meantime I have installed the guinea_pig over the stable. Domains now do get resolved when I use my router LAN IP address as the Custom DNS server within the WS client. Two points though:
> 
>     * nslookup still uses 127.0.0.1 as the resolver, this hides from the user the actual parent resolver used by Windscribe so I can only assume it uses my router. It would be nice if this can be displayed/confirmed anyhow.
> 
>     * how can I specify more than 1 Custom DNS server (e.g secondary DNS server available on the LAN) ? Would a comma (or space?) separated list work? Either ways this should be added as a comment on the GUI in my opinion.
> 
> 
> Thanks

Hi.  We have substantially changed how the custom DNS feature operates.  The Windscribe website is currently being overhauled to document this new functionality.  In a nutshell: when you set the `Connected DNS` setting to `Custom` and initiate the VPN connection, we spawn our ctrld (https://github.com/Control-D-Inc/ctrld) utility and instruct it to use the `Upstream 1` DNS server you specified.  This is why you are seeing 127.0.0.1 when using `nslookup`, as the `ctrld` utility is now servicing all DNS queries on your PC and forwarding them to your specified upstream DNS server.

If you want to specify an additional custom DNS server, enable the split DNS toggle.  This will allow you to specify a second upstream DNS server used by the ctrld utility and indicate which IPs/host names should use this secondary DNS server.

**Comment by wellloaded** [comment]
I just wanted to say that I have upgraded to the latest Beta on my windows laptop and I still experience this annoying DNS forced to 127.0.0.1/::1 issue.

As far as I can tell crtrld.exe is run when I specify a custom DNS "Connected DNS" so not ROBERT, if I eave ROBERT this process doesn't run. Regardless

![image](https://github.com/Windscribe/Desktop-App/assets/46318040/45cd016a-9a04-4cc1-b457-e35326dc7aaa)

I'm pretty much unable to change my interface DNS from the forced 127.0.0.1
![image](https://github.com/Windscribe/Desktop-App/assets/46318040/915795d6-20d0-4f04-9691-fe60fba2fcca)

Finally... this is something I really don't like!
![image](https://github.com/Windscribe/Desktop-App/assets/46318040/bdb688cf-17f0-4f1e-8d23-c95c9c6d60e2)

Please give us back full control on the local machine DNS settings!

**Comment by bernerdad** [comment]
Hi.  Are you saying that when specifying a custom DNS, you'd like the option to have the software behave the way it used to in the 2.6 version and not utilize the ctrld.exe DNS proxy?

**Comment by wellloaded** [comment]
Put in simple words: DNS on my device is an issue and doesn't work until I'm connected.
I understand why this might be wanted in certain cases but I think there should be a setting in the WS client like "DNS kill-switch" (so as defined in WS client or die) or something similar.

It is extremely frustrating to troubleshoot why my DNS points to 127.0.0.1/::1 and to be completely fair WS is just one of the players here, I can see a worrying trend in forcing DNS settings from software that think they are doing you a favor.

Bottom line please make sure that imposing DNS is clearly specified in the settings and especially simple to enable/disable via a single option.

Thanks! :-)
